### PR TITLE
ref(tally): use Bytes instead of u8 for the RevealResult

### PR DESCRIPTION
**Pull request by FranklinWaller**
## Motivation

Makes it simpler to work with Reveals

## Explanation of Changes

Convert `u8[]` to `Bytes` 

## Testing

`bun run test` works after a modification in the expected result (Bytes returns a different shape then u8[])

### CODE REVIEW

File: libs/as-sdk-integration-tests/assembly/tally.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { JSON } from 'json-as';
-import { Tally, Process, Bytes, OracleProgram } from '../../as-sdk/assembly';
+import { Tally, Process, Bytes, OracleProgram, Console } from '../../as-sdk/assembly';
```
Comment (Thomasvdam) [inline libs/as-sdk-integration-tests/assembly/tally.ts:2]: 🔥

File: libs/as-sdk-integration-tests/src/tallyvm.test.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ describe('TallyVm', () => {
 
     expect(result.exitCode).toBe(0);
     expect(result.resultAsString).toBe(
-      '[{"salt":[115,101,100,97,95,115,100,107],"exit_code":0,"gas_used":"200000","reveal":[123,34,100,97,116,97,34,58,34,98,97,98,121,95,115,104,97,114,107,34,125],"in_consensus":0},{"salt":[115,101,100,97,95,115,100,107],"exit_code":1,"gas_used":"1336","reveal":[123,34,100,97,116,97,34,58,34,103,114,97,110,100,112,97,95,115,104,97,114,107,34,125],"in_consensus":0}]'
+      '[{\"salt\":{\"type\":\"hex\",\"value\":\"736564615f73646b\"},\"exit_code\":0,\"gas_used\":\"200000\",\"reveal\":{\"type\":\"hex\",\"value\":\"7b2264617461223a22626162795f736861726b227d\"},\"in_consensus\":0},{\"salt\":{\"type\":\"hex\",\"value\":\"736564615f73646b\"},\"exit_code\":1,\"gas_used\":\"1336\",\"reveal\":{\"type\":\"hex\",\"value\":\"7b2264617461223a226772616e6470615f736861726b227d\"},\"in_consensus\":0}]'
```
Comment (Thomasvdam) [inline libs/as-sdk-integration-tests/src/tallyvm.test.ts:109]: Do the quotes need to be escaped?

**Review by Thomasvdam — APPROVED** [review]

File: libs/as-sdk-integration-tests/src/tallyvm.test.ts (hunk 2 above)
Comment (FranklinWaller) [reply libs/as-sdk-integration-tests/src/tallyvm.test.ts:109]: I just copied paste from expected

Comment (Thomasvdam) [reply libs/as-sdk-integration-tests/src/tallyvm.test.ts:109]: Ok ok I get it, you want Biome to do it for you :p

**Pull request merged**